src/lib/web/auth/oauth2/oauth2-custom-strategy.ts

```diff
@@ -15,14 +15,30 @@ function extractProfileAttribute (data, path: string): string {
 
 function parseProfile (data): Partial<PassportProfile> {
   const username = extractProfileAttribute(data, config.oauth2.userProfileUsernameAttr)
-  const displayName = extractProfileAttribute(data, config.oauth2.userProfileDisplayNameAttr)
-  const email = extractProfileAttribute(data, config.oauth2.userProfileEmailAttr)
+  let displayName: string | undefined
+  try {
+    // This may fail if the config.oauth2.userProfileDisplayNameAttr is undefined,
+    // or it is foo.bar and data["foo"] is undefined.
+    displayName = extractProfileAttribute(data, config.oauth2.userProfileDisplayNameAttr)
+  } catch (e) {
+    displayName = undefined
```

Comment:
Some debug logging that logging that no display name was provided would be nice.